mgr/manager.go

```diff
@@ -31,6 +31,60 @@ type CertServiceManager struct {
 	serviceManager svcmgr.Manager
 }
 
+func (csm *CertServiceManager) String() string {
+	return fmt.Sprintf("spec: %s", csm.Spec.Path)
+}
+
+// Process a spec updating content on disk, taking action as needed.
+// Returns (TTL for PKI, error).  If an error occurs, the ttl is at best
+// a hint to the invoker as to when the next refresh is required- that said
+// the invoker should back off and try a refresh.
+func (csm *CertServiceManager) EnforcePKI(enable_actions bool) (time.Duration, error) {
+	err := csm.CheckDiskPKI()
+	if err != nil {
+		log.Debugf("manager: %s, checkdiskpki: %s.  Forcing refresh.", csm, err.Error())
```

Comment:
Haven't compiled this and run it myself, but I'm curious why formatting `csm` as type `string` here isn't throwing an error

Reply:
(and in the rest of these log statements)

Reply:
I added a `String()` method to CertServiceManager. :)

That said, golang still would tolerate this at runtime if String() wasn't added- it just dumps an ugly struct output.

Our CI tests do check for this conversion error however, and they're passing.